modules/api/php/endpoints/candidate/visit/instrument/instrument.class.inc

```diff
@@ -171,7 +171,8 @@ class Instrument extends Endpoint implements \LORIS\Middleware\ETagCalculator
         try {
             $instrumentname = $this->_instrument->testName;
             $this->_instrument->clearInstrument();
-            $this->_instrument->_save($data[$instrumentname]);
+            $this->_instrument->_saveValues($data[$instrumentname]);
+            $this->_instrument->score();
```

Comment:
can I just move the data entry completion status logic to save values instead ? or you prefer not ?

Reply:
I looked into the IBIS and CCNA repositories and on IBIS there's 19 instrument classes that override `_saveValues`, mostly to implement instruments that have an uploader. (I think it might be the only way to do it?)

CCNA, on the other hand, has a bunch of scripts that call `_saveValues` but no instruments that override it.

It's probably safer to keep it where it is to ensure that it doesn't get missed.